Automattic/jetpack
# IDC: add blog ID to event tracking

#### Changes proposed in this Pull Request:
* add `assignSuperProps()` method to the Analytics class
* add blog ID to events tracked in IDC package
* fix a bug in `jetpack-analytics` package to properly track MC stats
* add `platform` value to Tracks

#### Does this pull request change what data or activity we track or use?
Adding blog ID to existing events.
No new events were introduced.

#### Testing instructions:
1. Bring the site into the Identity Crisis mode using the Jetpack Debug tools.
2. Open the browser's Network log and search by `idc`. Confirm that both Tracks event (`jetpack_idc_notice_view`) and MC stats event (`x_jetpack-idc=notice-view`) are being tracked.
3. Go to the MC "Tracks -> Live View" and search by the event `jetpack_idc_notice_view`. There'll likely be a delay of 5-15 minutes.
4. Confirm that it contains the property `platform=self-hosted`, and the `blogid` value in correct: <img width="318" alt="Screen Shot 2022-02-11 at 7 13 15 PM" src="https://user-images.githubusercontent.com/1341249/153687271-827b76ec-e89d-40ee-8e7b-a73341cc0ef6.png">
5. Add the constant to your WP config file: `define( 'WPCOM_IS_VIP_ENV', true );`, refresh the IDC screen with the browser's Network log open.
6. Confirm that the tracks event `jetpack_idc_notice_view` sends the property `platform=vip`.
